eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -274,16 +284,25 @@ object MultiPartHandler {
    * @param routes              routes that must be blinded and provided in the invoice.
    * @param router              router actor.
    * @param paymentPreimage_opt payment preimage.
+   * @param pathId_opt          path id for the payment paths. If None, a different one will be randomly generated for
+   *                            each path and the mapping blinding point -> path id will be store in the database.
+   * @param storeInDb           whether to store the incoming payment in the database. Bolt12 invoices generated as a
+   *                            response to an invoice request should not be stored as it would be a DoS vector.
    */
   case class ReceiveOfferPayment(nodeKey: PrivateKey,
                                  invoiceRequest: InvoiceRequest,
                                  routes: Seq[ReceivingRoute],
                                  router: ActorRef,
                                  paymentPreimage_opt: Option[ByteVector32] = None,
+                                 pathId_opt: Option[ByteVector] = None,
+                                 additionalTlvs: Set[InvoiceTlv] = Set.empty,
+                                 customTlvs: Set[GenericTlv] = Set.empty,
+                                 storeInDb: Boolean = true,
                                  paymentType: String = PaymentType.Blinded) extends ReceivePayment {
     require(nodeKey.publicKey == invoiceRequest.offer.nodeId, "the node id of the invoice must be the same as the one from the offer")
     require(routes.forall(_.nodes.nonEmpty), "each route must have at least one node")
     require(invoiceRequest.offer.amount.nonEmpty || invoiceRequest.amount.nonEmpty, "an amount must be specified in the offer or in the invoice request")
+    require(storeInDb || paymentPreimage_opt.nonEmpty, "the preimage will be lost")
```

Comment:
I don't understand this requirement, the preimage shouldn't be lost, if will be included in the payment metadata in our blinded path?

Reply:
The payment metadata has already been created at this point. If the preimage is not generated yet, then it can't be in the payment metadata.
Anyways, `paymentPreimage` and `pathId` are no longer optional.